packages/atomic/src/components/common/atomic-timeframe/atomic-timeframe.spec.ts

```diff
@@ -0,0 +1,154 @@
+import {html} from 'lit';
+import {describe, expect, it} from 'vitest';
+import {renderFunctionFixture} from '@/vitest-utils/testing-helpers/fixture';
+import type {AtomicTimeframe} from './atomic-timeframe';
+import './atomic-timeframe';
+
+describe('atomic-timeframe', () => {
+  const renderTimeframe = async ({
+    period = 'past',
+    unit = 'day',
+    amount = 1,
+    label,
+  }: {
+    period?: 'past' | 'next';
+    unit?: string;
+    amount?: number;
+    label?: string;
+  } = {}) => {
+    const element = await renderFunctionFixture(html`
+      <atomic-timeframe
+        period=${period}
+        unit=${unit}
+        amount=${amount}
+        label=${label ?? ''}
+      ></atomic-timeframe>
+    `);
+
+    const timeframe = element.querySelector(
+      'atomic-timeframe'
+    ) as AtomicTimeframe;
+
+    return {element, timeframe};
+  };
+
+  it('should have default period of "past"', async () => {
+    const {timeframe} = await renderTimeframe();
+    expect(timeframe.period).toBe('past');
+  });
+
+  it('should have default amount of 1', async () => {
+    const {timeframe} = await renderTimeframe();
+    expect(timeframe.amount).toBe(1);
+  });
+
+  it('should set period to "past"', async () => {
+    const {timeframe} = await renderTimeframe({period: 'past'});
+    expect(timeframe.period).toBe('past');
+  });
+
+  it('should set period to "next"', async () => {
+    const {timeframe} = await renderTimeframe({period: 'next'});
+    expect(timeframe.period).toBe('next');
+  });
+
+  it('should set unit to "day"', async () => {
+    const {timeframe} = await renderTimeframe({unit: 'day'});
+    expect(timeframe.unit).toBe('day');
+  });
+
+  it('should set unit to "month"', async () => {
+    const {timeframe} = await renderTimeframe({unit: 'month'});
+    expect(timeframe.unit).toBe('month');
+  });
+
+  it('should set unit to "year"', async () => {
+    const {timeframe} = await renderTimeframe({unit: 'year'});
+    expect(timeframe.unit).toBe('year');
+  });
```

Comment:
we should use it.each here and test all possible values (minute, hour, day, week, month, quarter, year).